async-download-rest/src/main/java/org/uniprot/api/async/download/messaging/consumer/MessageConsumer.java

```diff
@@ -100,9 +98,10 @@ public void onMessage(Message message) {
         }
     }
 
-    protected boolean isConsumedBefore(R downloadJob) {
-        return fileHandler.areAllFilesPresent(downloadJob.getId())
-                && ERROR != downloadJob.getStatus();
+    private static <R extends DownloadJob> boolean isJobStillRunning(R downloadJob) {
+        return downloadJob.getStatus() == RUNNING
+                || downloadJob.getStatus() == PROCESSING
+                || downloadJob.getStatus() == UNFINISHED;
     }
```

Comment:
isJobStillRunning doesn't capture the intent of checking that if the job is being run by another consumer/thread. We can use something like isJobRunning, isJobCurrentlyRunning, isJobRunningByOtherConsumer.
Do we need this method to be static? I think, we can remove it. I think, we can remove generics <R extends DownloadJob> because this method is being called from this onMessage method of this class only.